### PR TITLE
add vision API localize objects sample

### PR DESCRIPTION
**Pull request by dizcology**
add vision API localize objects sample

### CODE REVIEW

File: vision/localize_objects.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def localize_objects image_path:
+  # [START vision_localize_objects]
+  # image_path = "Path to local image file, eg. './image.png'"
+
+  require "google/cloud/vision"
+
+  vision = Google::Cloud::Vision::V1.new
```
Comment (beccca) [inline vision/localize_objects.rb]: `Google::Cloud::Vision.new`

It defaults to `Google::Cloud::Vision.new version: :V1`

For Alpha/Beta versions, you can provide `version: :VERSION`

File: vision/localize_objects.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  image_content = IO.read image_path, mode: 'rb'
```
Comment (beccca) [inline vision/localize_objects.rb]: `File.binread image_path`

File: vision/localize_objects.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  image = Google::Cloud::Vision::V1::Image.new content: image_content
```
Comment (beccca) [inline vision/localize_objects.rb]: Use Hashes rather than instantiating proto message objects

File: vision/localize_objects.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  feature = Google::Cloud::Vision::V1::Feature.new type: :OBJECT_LOCALIZATION
+
+  request = Google::Cloud::Vision::V1::AnnotateImageRequest.new image: image, features: [feature]
+
+  response = (vision.batch_annotate_images [request]).responses.first
```
Comment (beccca) [inline vision/localize_objects.rb]: Let's separate the `.responses.first` out explicitly

`batch_annotate_images` isn't necessarily intuitive

Comment (beccca) [reply vision/localize_objects.rb]: tl;dr The whole thing should come together looking something like:

```ruby
# image_path = "Path to local image file, eg. './image.png'"

require "google/cloud/vision"

vision = Google::Cloud::Vision.new

request = {
  features: [:OBJECT_LOCALIZATION],
  image: {
    content: File.binread(image_file)
  }
}

result  = vision.batch_attotate_images [request]
response = result.responses.first
objects  = response.localized_object_annotations

object.each do |onject|
# ...
```

**Comment by frankyn** [comment]
Deferring to @beccca for this review.

**Comment by beccasaurus** [comment]
*ping*

**Comment by beccasaurus** [comment]
~ _Happy Monday_ ~ `#ping`

File: vision/localize_objects.rb (hunk 1 above)
Comment (dizcology) [reply vision/localize_objects.rb]: with this change, the tests failed with the following error:
```
NoMethodError:
       undefined method `batch_annotate_images' for #<Google::Cloud::Vision::Project:0x00007feb084a4c50>
```

File: vision/localize_objects.rb (hunk 2 above)
Comment (dizcology) [reply vision/localize_objects.rb]: done

File: vision/localize_objects.rb (hunk 3 above)
Comment (dizcology) [reply vision/localize_objects.rb]: done

File: vision/localize_objects.rb (hunk 4 above)
Comment (dizcology) [reply vision/localize_objects.rb]: done

File: vision/localize_objects.rb (hunk 1 above)
Comment (beccasaurus) [reply vision/localize_objects.rb]: https://github.com/googleapis/google-cloud-ruby/issues/2482

Accepted as a feature request

Comment (beccasaurus) [reply vision/localize_objects.rb]: PR sent and pending LGTM & merge & publish new gem
https://github.com/googleapis/google-cloud-ruby/pull/2487#pullrequestreview-159102063

**Pull request merged**